### DISCUSSION TITLE
MagickCacheErrorException Image Magick no pixels defined in cache @ error/cache/OpenPixelCache/[number here]

### DISCUSSION THREAD

**Discussion by StealthRT** [ImageMagick/ImageMagick#4683] [Help]
Hey all I am getting this error when running my program. It starts out fine but once the 4th loop comes around it gives the error as seen below:

https://i.stack.imgur.com/7jKLv.png
![QGfqT](https://user-images.githubusercontent.com/1037844/148701648-02f3cb46-896e-4126-8904-8667c091f6a0.png)

> ImageMagick.MagickCacheErrorException: 'no pixels defined in cache `' @ error/cache.c/OpenPixelCache/3671'

Affected code:

    MagickImage circleText = new MagickImage(MagickColors.Transparent, 60, 60);
    MagickImage circle = new MagickImage(MagickColors.Transparent, 70, 70);
    MagickImage _shadow = new MagickImage();
    
    circleText.Settings.StrokeColor = MagickColors.Black;
    circleText.Settings.FontPointsize = 18;
    circleText.Settings.FontWeight = FontWeight.UltraBold;
    circleText.BackgroundColor = MagickColors.Transparent;
    circleText.Settings.FillColor = MagickColors.White;
    circleText.Settings.TextGravity = Gravity.Center;
    circleText.Settings.TextAntiAlias = true;
    circleText.Settings.FontFamily = "Arial";
    circleText.Settings.StrokeAntiAlias = true;
    circleText.Settings.StrokeWidth = 1;
    circleText.Read("label:+" + subtractImgs.ToString(), 50, 40);
    circleText.Resize(50, 80);
    
    circle.Draw(new DrawableStrokeColor(MagickColors.Black),
                new DrawableStrokeWidth(1),
                new DrawableFillColor(MagickColors.Red),
                new DrawableCircle(25, 20, 20, 3));
    
    _shadow = new MagickImage("xc:none", 60, 60);
    _shadow.Settings.FillColor = MagickColors.Black;
    _shadow.Draw(new DrawableCircle(25, 20, 20, 3));
    _shadow.Blur(0, 2);
    
    circle.Composite(_shadow, CompositeOperator.DstOver);
    
    circleText.Composite(circle, CompositeOperator.DstOver);
    
    result.Composite(circleText, 375, 6, CompositeOperator.Over);
    result.Write(path + "directoryhere\\" + item.id + ".png");
    
    roundImg.Dispose();
    mask.Dispose();
    shadow.Dispose();
    result.Dispose();
    GC.Collect();

I'm using Mahick.net on version 7.17.0.1. I'm not on the latest due to any version over 7.17.0.1 breaks a lot of my imagemagick code.

It seems to have been fixed in version 7.11.0.0 according to this post but I'm still having the issue despite being a good version up from the fixed version?

What would I need to correct this issue?

**Comment by snibgo** [comment]
> ... but once the 4th loop comes around ...

I can't see a loop in your code.

You have:
```
MagickImage _shadow = new MagickImage();
:
:
_shadow = new MagickImage("xc:none", 60, 60);
```
Is that wise?

You have:
```
result.Composite(circleText, 375, 6, CompositeOperator.Over);
```
... but no code that defines "result" or defines any pixels in it.

**Comment by StealthRT** [comment]
It seems the image it has is causing "result" to say **Unknown 0x0 16-bit sRGB**
![2022-01-09 22_24_51-DataGatheringForFB (Debugging) - Microsoft Visual Studio](https://user-images.githubusercontent.com/1037844/148715077-a9dca3e4-5103-4797-94f8-91e9c4e5d5f9.png)

While the 3 images before it says:
![2022-01-09 22_26_49-DataGatheringForFB (Debugging) - Microsoft Visual Studio](https://user-images.githubusercontent.com/1037844/148715187-51249fa1-1cfc-4993-8d52-44d053fc3a33.png)

I also noticed that the image is larger than the previous 3 images. The previous images were 720x960 and this one causing an issue is 960x960.

What would cause this?

**Comment by snibgo** [comment]
Your command...
```
result.Composite(circleText, 375, 6, CompositeOperator.Over);
```
... tries to modify pixels in the image named "result". If that image doesn't already have pixels, then that is the problem.

**Comment by StealthRT** [comment]
Ah found the reason. The image was 960px x 960px so some of the math i was doing caused the end result to be 0px. Thanks for the help @snibgo
